src/key.js

```diff
@@ -771,6 +771,92 @@ Key.prototype.verifyAllUsers = async function(keys) {
   return results;
 };
 
+/**
+ * generate and add a new subkey to a Key
+ * @param {String}  options.passphrase (optional) The passphrase used to encrypt the resulting private subkey and decrypt the security key
+ * @param  {Boolean} options.sign (optional) defaults to false, and indicates whether the subkey should sign rather than encrypt
+ * @param {Number} [options.keyExpirationTime=0]
+ *                             The number of seconds after the key creation time that the key expires
+ * @param {Integer} options.numBits (optional) number of bits for the key creation.
+ * @param  {String} options.curve (optional) elliptic curve for ECC keys
+ * @param  {Date} options.date  Override the creation date of the key and the key signatures
+ * @param  {Object} userId                   (optional) user ID
+ * @returns {Promise<module:key.SubKey>} return the subkey if successful.
+ * @async
+ */
+Key.prototype.generateSubkey = async function(options){
+  const defaultOptions = {};
+  const defaultAlgo = this.primaryKey.algorithm;
+  if (defaultAlgo.indexOf('rsa') === 0) {
+    defaultOptions.numBits = this.primaryKey.params[0].bitLength();
+  }
+  else if (defaultAlgo.indexOf('ec') === 0 || defaultAlgo.indexOf('ed') === 0) {
```

Comment:
Could you instead check explicitly for ecdsa, eddsa, ecdh? In case there is some conflict in the future with algorithms that coincidentally have these prefixes.